src/test/java/org/jabref/logic/integrity/IntegrityCheckTest.java

```diff
@@ -390,21 +84,21 @@ private BibDatabaseContext createContext(Field field, String value, EntryType ty
         return new BibDatabaseContext(bibDatabase);
     }
 
-    private BibDatabaseContext createContext(Field field, String value, MetaData metaData) {
+    protected static BibDatabaseContext createContext(Field field, String value, MetaData metaData) {
```

Comment:
Why was the modifier changed? Is this method used outside of this class? It should be `private`. Adding `static` is OK for me.

Reply:
@koppor yes it was used outside of the class. I changed it now so all classes are independent. And I think I covered all the issues mentioned with my new commits. Please check it out and let me know what you think!